Composer/packages/form-dialogs/src/undo/UndoRoot.tsx

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
Nit: UndoRoot reads like a command.  RecoilRootWithUndo?  UndoableRecoilRoot? There may not be a better name...

Reply:
I think it's good enough.